### PR TITLE
Added support for a fully loaded, scrollable, contained ladder.
If you want to disable it, reload the page or whatever.

### PR DESCRIPTION
**Pull request by LynnCinnamon**

### CODE REVIEW

File: fairgame.js [hunk 1]
```diff
@@ -390,8 +390,29 @@ function setLadderRows() {
...
+    ladderContainer.style.height = "64vh";
+    ladderContainer.style.overflow = "auto";
+    ladderContainer.style.border = "gray solid 2px";
+    ladderParent.replaceChild(ladderContainer, ladder);
+    ladderContainer.appendChild(ladder);
+}
+
 addJS_Node(secondsToHms);
 addJS_Node(solveQuadratic);
+addJS_Node(expandLadder);
```
Comment (Quadrum1) [inline fairgame.js:418]: Can you add an option to call expandLadder() automatically?
if (qolOptions.scrollableLadder) {expandLadder()}

Comment (Quadrum1) [reply fairgame.js:418]: Would need to add that config at the top as well then

File: fairgame.js [hunk 2]
```diff
@@ -390,8 +390,29 @@ function setLadderRows() {
     clientData.ladderPadding = qolOptions.expandedLadder.size / 2;
 }
 
+
+function expandLadder() {
+    if(document.getElementsByClassName("ladder-container").length > 0) {
+        return;
+    }
+    var input = 10000;
```
Comment (Quadrum1) [inline fairgame.js]: This could lead to some client lag, better keep it at 250-500 or so?

File: fairgame.js [hunk 3]
```diff
@@ -413,6 +434,7 @@ $("#offcanvasOptions").children(".offcanvas-body").html(`<div style="display: bl
                        <option value="Lato">Lato</option>
                        </select></div>`+
                        `<div style="display: block; padding: 0.5rem; font-size:1.25rem"><span>Ladder Rows</span><div class="input-group"><input class="form-control shadow-none" id="rowsInput" maxlength="2" placeholder="# of rows, min 10" type="text"><button class="btn btn-primary shadow-none" id="rowsButton" onclick="setLadderRows()">Set</button></div></div>`+
+                       `<button class="btn btn-primary shadow-none" id="rowsButton" onclick="expandLadder()">Full, scrollable Ladder</button>`+
```
Comment (Quadrum1) [inline fairgame.js]: Having two buttons with id "rowsButton" could lead to issues.

File: fairgame.js (hunk 2 above)
Comment (LynnCinnamon) [reply fairgame.js]: 250 -500 would not give us the entire ladder.
It should only lag if we have actually that many rankers. I will add an option for the input size.

File: fairgame.js [hunk 4]
```diff
@@ -390,8 +394,28 @@ function setLadderRows() {
     clientData.ladderPadding = qolOptions.expandedLadder.size / 2;
 }
 
+
+function expandLadder() {
+    if(document.getElementsByClassName("ladder-container").length > 0) {
+        return;
+    }
+    qolOptions.expandedLadder.size = qolOptions.scrollableLadder.size;
```
Comment (Quadrum1) [inline fairgame.js:402]: 1 = 1
:smile:

**Review by Quadrum1 — APPROVED** [review]

**Pull request merged**